src/collective/solr/manager.py

```diff
@@ -1,67 +1,23 @@
 # -*- coding: utf-8 -*-
-from collective.solr.interfaces import ISolrConnectionConfig
 from collective.solr.interfaces import ISolrConnectionManager
 from collective.solr.interfaces import IZCMLSolrConnectionConfig
 from collective.solr.local import getLocal
 from collective.solr.local import setLocal
 from collective.solr.solr import SolrConnection
+from collective.solr.utils import getConfig
+from collective.solr.utils import isActive
 from httplib import CannotSendRequest
 from httplib import ResponseNotReady
 from logging import getLogger
-from persistent import Persistent
 from socket import error
-from zope.component import getUtility
 from zope.component import queryUtility
 from zope.interface import implements
+from plone.api.portal import get_registry_record
 
 logger = getLogger('collective.solr.manager')
 marker = object()
 
 
-class BaseSolrConnectionConfig(object):
```

Comment:
@tomgross in theory yes, in practice no. Things are way more complex than that. By removing the ISolrConnectionConfig you also removed the IZCMLSolrConnection config lookup. This removal requires a considerable refactoring both in the code and in the test setup. 

I tried to mock the registry lookup for the unit tests, but this did not work out... :(